src/number_theory/function_field.lean

```diff
@@ -197,6 +200,62 @@ begin
   rw [infty_valuation_def, if_neg hp', ratfunc.int_degree_polynomial]
 end
 
+/-- The valued field `Fq(t)` with the valuation at infinity. -/
+def infty_valued_Fqt : valued (ratfunc Fq) (with_zero (multiplicative ℤ)) :=
+⟨infty_valuation Fq⟩
+
+lemma infty_valued_Fqt.def {x : ratfunc Fq} :
+  @valued.v (ratfunc Fq) _ _ _ (infty_valued_Fqt Fq) (x) = infty_valuation_def Fq x := rfl
+
+/-- The topology structure on `Fq(t)` induced by the valuation at infinity. -/
+def tsq' : topological_space (ratfunc Fq) :=
+@valued.topological_space (ratfunc Fq) _ _ _ (infty_valued_Fqt Fq)
+
+lemma tdrq' : @topological_division_ring (ratfunc Fq) _ (tsq' Fq) :=
+@valued.topological_division_ring (ratfunc Fq) _ _ _ (infty_valued_Fqt Fq)
+
+/-- The uniform structure on `k(t)` induced by the valuation at infinity. -/
+def usq' : uniform_space (ratfunc Fq) :=
+@topological_add_group.to_uniform_space (ratfunc Fq) _ (tsq' Fq) _
+
+lemma ugq' : @uniform_add_group (ratfunc Fq) (usq' Fq) _ :=
+@topological_add_group_is_uniform (ratfunc Fq) _ (tsq' Fq) _
+
+lemma cfq' : @completable_top_field (ratfunc Fq) _ (usq' Fq) :=
+@valued.completable (ratfunc Fq) _ _ _ (infty_valued_Fqt Fq)
+
+lemma ssq' : @separated_space (ratfunc Fq) (usq' Fq) :=
+@valued_ring.separated (ratfunc Fq) _ _ _ (infty_valued_Fqt Fq)
```

Comment:
These names are somewhat nonstandard. Would you please write them out?
I guess the `'` is to avoid conflicts with the instances below? Maybe one or the other could be wrapped in a namespace?

Reply:
Done.